src/main/java/main/Duke.java

```diff
@@ -0,0 +1,189 @@
+            }
+            t = list.get(index - 1);
+            t.isDone =  false;
+            System.out.println("OK, I've marked this task as not done yet:");
+        }
+
+        System.out.println(t);
+        System.out.println("____________________________________________________________");
+    }
+
+    // Function to print the list of tasks
+    public static void userList(String line, ArrayList<Task> list) {
+            for (Task task : list) {
+                if (task == null) {
+                    break;
+                }
+                System.out.println(task.index + ". " + task);
+            }
+            System.out.println("____________________________________________________________");
+    }
+
+    // Function to say bye
+    public static void userBye() {
+        System.out.println("Bye human. Come back soon !");
+    }
+
+    // Function to print an unknown command from user
+    public static void userWrongCommand() {
+        System.out.println("No suitable command found. Please try again!");
+    }
+
+    // Function to check the minimum arguments supplied for each task
+    public static boolean checkMinimumArguments(String[] splitLine, int number) {
+        try {
+            if (splitLine.length < number) {
+                throw new DukeException("Invalid Syntax! Please try again!");
+            }
+        } catch (DukeException e) {
+            System.out.println("Minimally " + number + " arguments, please try again!");
+            return true;
+        }
+        return false;
+    }
+
+    // Start of user input
+    public static void userInput() throws DukeException {
+        Scanner scanner = new Scanner(System.in); // Declared a scanner object
+        String line; // Declared a string object to take in user input
+        ArrayList<Task> list = new ArrayList<>(); // Declared a dynamic list with Task data type
+        int counter = 0;
+        int index = 0;
+        Task t = null;
+
+        // Start of user input
+        while (true) {
+            line = scanner.nextLine().toLowerCase(); // Takes in user input
+            String[] splitLine = line.split("\\s+"); // split by whitespaces
+            String command = splitLine[0]; //obtain the main command from user, which is the first command
+
+            switch (command) {
+                // User wants to exit
+                case "bye":
+                    userBye();
+                    break;
+
+                // User wants to display the list of tasks
+                case "list":
+                    if (splitLine.length != 1) {
+                        userWrongCommand();
+                        break;
+                    }
+                    userList(line, list);
+                    continue;
+
+                // User wants to mark or unmark tasks
+                case "mark":
+                case "unmark":
+                    userMarkOrUnmark(splitLine[0], line, list, index, t);
+                    continue;
```

Comment:
Ensure that a fallthrough comment should be included if there are no break statements for `case` clause